### PR TITLE
Update mcr.cpp

### PR DESCRIPTION
**Pull request by Yitong-web**

### CODE REVIEW

File: mcr.cpp [hunk 1]
```diff
@@ -23,19 +26,22 @@ bool isWin(char game[3][3]){
...
-    for (int n=0; n<9; n++){ 
+    for (n=0; n<9; n++){ 
         turn = !turn;  // use the not-operator to change true to false or false to true. 
         if (turn == false) 
             cout << "Player 1: "; 
         else 
             cout << "Player 2: "; 
-        cout << "Which cell to mark? i:[1..3], j:[1..3]: ";  
-        cin >> i >> j; 
+        do {
```
Comment (Kunz1Pro) [inline mcr.cpp:41]: good change

Comment (winnerFG) [reply mcr.cpp:41]: it really make sense  excellent

File: mcr.cpp [hunk 2]
```diff
@@ -23,19 +26,22 @@ bool isWin(char game[3][3]){
  
 int main(){ 
     int i, j; 
-    char game[3][3] = {' '}; // Tic-tac-toe
```
Comment (PeiCiYee) [inline mcr.cpp:26]: this cannot initialize the array, need some changes

File: mcr.cpp [hunk 3]
```diff
@@ -23,19 +26,22 @@ bool isWin(char game[3][3]){
...
     bool turn = true; // false for player 1's turn, true for player 2's turn. Player 1 first. 
     cout << "X = Player 1" << endl << "O = Player 2" << endl; 
-    for (int n=0; n<9; n++){ 
+    for (n=0; n<9; n++){ 
         turn = !turn;  // use the not-operator to change true to false or false to true. 
         if (turn == false) 
             cout << "Player 1: "; 
         else 
             cout << "Player 2: "; 
-        cout << "Which cell to mark? i:[1..3], j:[1..3]: ";
```
Comment (PeiCiYee) [inline mcr.cpp:37]: The range for the i and j should be 0 to 2 instead of 1 to 3

File: mcr.cpp [hunk 4]
```diff
@@ -23,19 +26,22 @@ bool isWin(char game[3][3]){
...
         turn = !turn;  // use the not-operator to change true to false or false to true. 
         if (turn == false) 
             cout << "Player 1: "; 
         else 
             cout << "Player 2: "; 
-        cout << "Which cell to mark? i:[1..3], j:[1..3]: ";  
-        cin >> i >> j; 
+        do {
+                cout << "Which cell to mark? i:[0..2], j:[0..2]: ";
+                cin >> i >> j;
```
Comment (zw-954997) [inline mcr.cpp:43]: Here we should determine whether the values of i and j are valid (whether they are between 0 and 2)

File: mcr.cpp [hunk 5]
```diff
@@ -23,19 +26,22 @@ bool isWin(char game[3][3]){
...
+    int n;
+    char game[3][3] = {{' ', ' ', ' '}, {' ', ' ', ' '}, {' ', ' ', ' '}}; // Tic-tac-toe
     char player1 = 'X'; 
     char player2 = 'O'; 
     bool turn = true; // false for player 1's turn, true for player 2's turn. Player 1 first. 
     cout << "X = Player 1" << endl << "O = Player 2" << endl; 
-    for (int n=0; n<9; n++){ 
+    for (n=0; n<9; n++){ 
         turn = !turn;  // use the not-operator to change true to false or false to true. 
         if (turn == false)
```
Comment (Pan-yancheng) [inline mcr.cpp:37]: It's not concise enough

**Pull request closed without merging**